.github/workflows/ci.yml

```diff
@@ -63,14 +41,14 @@ jobs:
         fetch-depth: 0
 
     - name: Set up Go
-      uses: actions/setup-go@v4
+      uses: actions/setup-go@v5
       with:
-        go-version: '1.21'
+        go-version: '1.23'
```

Comment:
You can remove the with here, it will use the last stable, it's enough for goreleaser I think, and so no need to maintain the latest go version here.

Another solution is to use 'go.mod' as a value here, please double check by reading the documentation of the GitHub action